text/0017-message/0017-message.md

```diff
@@ -0,0 +1,169 @@
+
+# Summary
+
+A message is the object nodes gossip around in the network. It always references two other messages that are known as `parents`. It is stored as a vertex on the tangle data structure that the nodes maintain.
+
+The messages will contain payloads. Some of them will be core payloads that will be processed by all nodes as part of the core protocol. Some of them will be community payloads that will enable the building of new functionalities on top of the Tangle. Some payloads may have other nested payloads embedded inside.
+So upon parsing, it is done layer by layer.
+
+# Motivation
+
+To better understand this layered design, consider the internet protocol, for example: there is an Ethernet frame that contains an IP payload. This in turn contains a TCP packet that encapsulates an HTTP payload. Each layer has a certain responsibility and once this responsibility is completed, we move on to the next layer.
+
+The same is true with how messages are parsed. The outer layer of the message enables the mapping of the message to a vertex in the Tangle and performing some basic validation. The next layer may be a transaction that mutates the ledger state, and the next layer may provide some extra functionality on the transactions to be used by applications.
+
+By making it possible to add and exchange payloads, an architecture is being created that can be easily extended to accommodate future needs.
+
+# Detailed design
+
+### Data types
+
+The following are data types that we will use when we specify fields in the message and payloads.
+
+| Name   | Description   |
+| ------ | ------------- |
+| uint8  | An unsigned 8 bit integer encoded in Little Endian. |
+| uint16  | An unsigned 16 bit integer encoded in Little Endian. |
+| uint32  | An unsigned 32 bit integer encoded in Little Endian. |
+| uint64  | An unsigned 64 bit integer encoded in Little Endian. |
+| ByteArray[N] | A static size array of size N.   |
+| ByteArray | A dynamically sized array. A uint32 denotes its length.   |
+| string | A dynamically sized array of an UTF-8 encoded string. A uint16 denotes its length.   |
+
+
+### Message ID
+The message ID will be the `BLAKE2b-256` hash of the byte contents of the message. It should be used by the nodes to index the messages and by external APIs.
+
+
+### Message structure
+
+<table>
+    <tr>
+        <th>Name</th>
+        <th>Type</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td>NetworkID</td>
+        <td>uint64</td>
+        <td>Network identifier. This field will signify whether this message was meant for mainnet, testnet, or a private net. It also tells what protocol rules apply to the message. It is first 8 bytes of the `BLAKE2b-256` hash of the concatenation of the network type and the protocol version string.</td>
+        </tr>
+    <tr>
+        <td> Parents' length </td>
+        <td> uint8</td>
+        <td> The number of messages we directly approve. Can be any value between 1-8.</td>
```

Comment:
Could you clarify the exact range here a little bit? It's not immediately obvious if "between 1-8" is inclusive or exclusive (or half open), and in the first sentence of the summary it says "It always references two other messages that are known as `parents`".

Maybe something like: "Can be any value in the range (1,8]"?